source/tests/system/nixnet_lin_driver_api_tests.cpp

```diff
@@ -123,7 +143,7 @@ TEST_F(NiXnetLINDriverApiTests, FrameStreamInputFromExample_FetchData_DataLooksR
   EXPECT_SUCCESS(client::clear(stub(), session));
 }
 
-TEST_F(NiXnetLINDriverApiTests, LoopbackTestFromExample_FetchData_DataLooksReasonable)
+TEST_F(NiXnetLINDriverApiTestsWithHardware, LoopbackSignalSinglePointTestFromExample_WriteAndReadSignal_SignalMatches)
```

Comment:
This note would be a good code comment.

Does it work to read until the value is not zero? Like the `WaitForAsync` asserts in ASW? I don't think that's necessary, and often doesn't work with HW tests. Just curious. And it's a good best practice to avoid sleeps if possible.

Reply:
It does and typically it took 6 to 8 read API calls in tight loop for me to get valid signal. Would it be better if I implement tight loop which tries 20 times before failing?

Other option would be ASW equivalent timeout approach, but that seems costlier for single test.

Test time came down from ~1 second to 76ms after adding counter based loop.

Reply:
Good question!

1. It's probably better to throttle it with 1-10ms sleeps (for lower-level integration tests I would say 1ms; I would lean 10ms in this case).
2. I don't think you need to make your criteria "stricter" than allowing 1 second. We're not interested in verifying any timing expectation. Actually it's the opposite: we want to make as few timing assumptions as possible. 

So, for example, you could loop up to 100 times with a 10ms sleep and exit the loop early if you hit the expected value (or any non-zero value). Then (a) the test still finishes ~fast, (b) we've got minimal risk of a false failure on a slow system and (c) we're (hopefully) not putting the system/driver under abnormal load.

I think that would be better; what you have is OK too. Thanks for the follow up.

Reply:
Done.